src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4110,19 +4118,19 @@ private ImmutableArray<IPropertySymbol> GetIndexerGroupSemanticSymbols(BoundExpr
             }
 
             Binder binder = binderOpt ?? GetEnclosingBinder(GetAdjustedNodePosition(boundNode.Syntax));
-            var symbols = ArrayBuilder<ISymbol>.GetInstance();
```

Comment:
>var symbols = ArrayBuilder<ISymbol>.GetInstance(); [](http://example.com/codeflow?start=12&length=50)

Do we have an evidence that this code path has the same problem as ```LookupSymbolsInternal```? I can see how ```LookupSymbolsInternal``` could have thousands of items in the result when ```name``` is not specified. In this case, however, it looks like we are dealing with indexers in a specific type, there aren't usually that many of them.

Reply:
No, I have not seen this method in the profiles I was looking at. I was trying to be consistent with the other change away from ArrayBuilder<ISymbol> in this file, but I can change back to the original here if you prefer.

Reply:
>  I was trying to be consistent with the other change away from ArrayBuilder<ISymbol> in this file

I do not think we should be making changes based solely on this motive

Reply:
Not reverting this one (yet) as it's a bit more entwined with the changes needed for LookupSymbolsInternal.

This method calls AppendSymbolsWithNameAndArity with the symbol list. 

LookupSymbolsInternal calls AppendSymbolsWithName which calls AppendSymbolsWithNameAndArity with the symbol list.

Changing one of these to SegmentedList but not the other would require duplicating AppendSymbolsWithNameAndArity to be implemented against both SegmentedList and ArrayBuilder (or just against a common base, but ArrayBuilder only implements IReadOnly interfaces).

Is that good enough justification to keep this part of the change, or would you prefer for it to be changed back to the original at the tradeoff of needing to (likely) duplicate at least some part of AppendSymbolsWithNameAndArity?

Reply:
Maybe iteration 5 is preferable? It didn't require duplicating code and continues to use ArrayBuilder at this location.

Reply:
> Maybe iteration 5 is preferable? It didn't require duplicating code and continues to use ArrayBuilder at this location.

If we are sharing a helper with ```LookupSymbolsInternal``` (sorry I didn't realize that before) that is good enough reason to use the same data structure here as well. I would go back to the iteration 3 version of this method and would get rid of the delegate.